apps/web/cypress/tests/templates-store.spec.ts

```diff
@@ -0,0 +1,192 @@
+describe('Templates Store', function () {
+  const getTemplateDetails = (templateName: string): { name: string; iconName: string } => {
+    const regexResult = /^:.{1,}:/.exec(templateName);
+    let name = '';
+    let iconName = 'fa-solid fa-question';
+    if (regexResult !== null) {
+      name = templateName.replace(regexResult[0], '').trim();
+      iconName = regexResult[0].replace(/:/g, '').trim();
+    }
+
+    return { name, iconName: iconName };
+  };
+
+  beforeEach(function () {
+    cy.initializeSession({ noTemplates: true }).as('session');
+    indexedDB.deleteDatabase('localforage');
+  });
+
+  it('the all templates tile should be disabled when there are no blueprints ', function () {
+    cy.intercept('**/notification-templates**').as('getTemplates');
+    cy.intercept('**/blueprints/group-by-category').as('getBlueprints');
+    cy.visit('/workflows');
+    cy.wait('@getTemplates');
+
+    cy.getByTestId('no-workflow-templates-placeholder').should('be.visible');
+    cy.getByTestId('create-workflow-tile').should('exist');
+    cy.getByTestId('all-workflow-tile').should('exist').should('be.disabled');
+  });
+
+  it('the all templates tile should be enabled and popular should be shown when blueprints are fetched', function () {
+    cy.makeBlueprints();
+    cy.intercept('**/notification-templates**').as('getTemplates');
+    cy.intercept('**/blueprints/group-by-category').as('getBlueprints');
+    cy.visit('/workflows');
+    cy.wait('@getTemplates');
+
+    cy.getByTestId('no-workflow-templates-placeholder').should('be.visible');
+    cy.getByTestId('create-workflow-tile').should('exist');
+    cy.getByTestId('all-workflow-tile').should('exist').should('be.disabled');
+
+    cy.wait('@getBlueprints');
+    cy.getByTestId('all-workflow-tile').should('exist').should('not.be.disabled');
+    cy.getByTestId('popular-workflow-tile').should('have.length', 3);
```

Comment:
This one is correct but at the moment we have only 2 popular templates, therefore I will update all the tests to check length of 2.